exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillPreparedStatementImpl.java

```diff
@@ -259,4 +261,17 @@ public void setObject(int parameterIndex, Object x, SQLType targetSqlType) throw
     checkOpen();
     super.setObject(parameterIndex, x, targetSqlType);
   }
+
+  @Override
+  public void setLargeMaxRows(long maxRowCount) throws SQLException {
+    Statement setMaxStmt = this.connection.createStatement();
+    setMaxStmt.execute("ALTER SESSION SET `" + ExecConstants.QUERY_MAX_ROWS + "`=" + maxRowCount);
+    setMaxStmt.close();
+    this.maxRowCount = maxRowCount;
```

Comment:
Please remove this line and put into the beginning of the method call `super.setLargeMaxRows(maxRowCount)`, so it will check that connection is open and validate `maxRowCount` value.

The same for changes in `DrillStatementImpl`.

Reply:
Ok, but I think it should be done post-close() because in case of a failure, we dont want to accidentally set through `super` and then worry about rolling back.

Reply:
It is important to check that connection is opened at first, and it is done in the parent method.

Reply:
👍